accord-core/src/main/java/accord/topology/TopologyManager.java

```diff
@@ -231,8 +254,10 @@ static class Notifications
             Ranges closed = Ranges.EMPTY, retired = Ranges.EMPTY;
         }
 
-        private static final Epochs EMPTY = new Epochs(new EpochState[0]);
+        private static final Epochs EMPTY = new Epochs(new EpochState[0], new ArrayList<>(), new ArrayList<>(), -1);
         private final long currentEpoch;
+        private final long firstNonEmptyEpoch;
```

Comment:
do we need to precompute this? can we calculate it on the fly? I think we don't query it often

Reply:
talked off-github: we can't avoid precomputing this field, since it is a "historical" information. In other words, we can not infer it form `Epochs`. 

Tried moving it to a different class, but unfortunately alternatives are only worse than the current version.